unity-renderer/Assets/Desktop/Scripts/MainScripts/DCL/Controllers/HUD/Preloading/PreloadingController.cs

```diff
@@ -13,34 +15,40 @@ public class PreloadingController : IDisposable
         private readonly DataStoreRef<DataStore_LoadingScreen> loadingScreenRef;
 
         private BaseVariable<bool> isSignUpFlow => DataStore.i.common.isSignUpFlow;
-        private bool isDisposed = false;
+        private bool isDisposed;
 
-        public PreloadingController()
+        private CancellationTokenSource cancellationTokenSource;
+        public PreloadingController(IAddressableResourceProvider addressableResourceProvider)
         {
-            view = Object.Instantiate(GetView());
+            cancellationTokenSource = new CancellationTokenSource();
+            GetView(cancellationTokenSource.Token).Forget();
+
             loadingScreenRef.Ref.loadingHUD.message.OnChange += OnMessageChange;
             loadingScreenRef.Ref.decoupledLoadingHUD.visible.OnChange += OnDecoupledLoadingScreenVisibilityChange;
             isSignUpFlow.OnChange += SignUpFlowChanged;
-        }
 
-        private void OnDecoupledLoadingScreenVisibilityChange(bool current, bool _)
-        {
-            if (current)
-                Dispose();
-        }
-
-        private GameObject GetView()
-        {
-            return Resources.Load<GameObject>("Preloading");
+            async UniTask GetView(CancellationToken cancellationToken) =>
+                view = (await addressableResourceProvider.Instantiate<Transform>("Preloading", cancellationToken: cancellationToken))
```

Comment:
This is not a change request, but more of something to add to our code conventions. I ask for the async asset and then instantiate, while you are using directly iasync nstantiate. Either to me is fine, lets chose one and stick to it :).

Reply:
I am fine with that until we don't call `Addressables.InstantiateAsync(address)` multiple times for the same asset

Reply:
I would say that there are different use cases. One mentioned by Mikhail, when you need to load once and instantiate several times. Another case could be when you need to `Initialize` before the `Awake` (because as I remember, `InstantiateAsync` calls `Awake` on prefab components before any of your custom calls)

Reply:
>Another case could be when you need to Initialize before the Awake

I didn't get this case.
It should be just the same as for the usual `Instantiate`

>before any of your custom calls

What kind of calls do you mean?
In case you have the synchronous piece like:
```
Instantiate(...) // will call `Awake`, then `Start` if the object is active
// calls to the object

```

Reply:
No, it is not the same. In `Addressables.InistantiateAsync` there is this special thing (which often makes transition from `Resources.InstantiateAsync` to it more difficult), because Addressables calls `Awake` method before any of your custom code. You can make simple experiment by yourself and see that with this in the console (Awake before Initialize)
```
var view =  await new AddressableResourceProvider().Instantiate<LoadingHUDView>("LoadingHUD");
view.Initialize();
```
and logs inside `LoadingHUDView` `Awake`, `Start` and `Initialize` methods, you will get these logs
![image](https://user-images.githubusercontent.com/35366872/218489551-c44c4527-2ae8-4d30-8357-093e99e9d73e.png)